### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by NeetishPathak**
*  Changes 
- Updating deprecated methods ioutil.ReadFile,ioutil.ReadAll and ioutil.WriteFile to os.ReadFile,io.ReadAll and os.WriteFile respectively
-  Using err.Error() to get Error String
- Using errors.Is with the appropriate error type
- Remove bad nil guard
- use net.JoinHostPort instead of fmt.Sprintf(, ) - best practice
- Cleanup certs_test.go

* the motivation for the change - are to include best practices and fix security recommendations

### CODE REVIEW

**Review by jostanislas — DISMISSED** [review]
Looks Good to Me.

File: cmd/storageserv/config/config.go [hunk 1]
```diff
@@ -163,7 +165,7 @@ func LoadConfig(ssConfigFile string) (err error) {
 	if !serverConfig.MicroShardsEnabled && len(serverConfig.DB.DbPaths) >= 1 {
 		tagFile := fmt.Sprintf("%s/microshard_enabled.txt", serverConfig.DB.DbPaths[0].Path)
 		_, err := os.Stat(tagFile)
-		if !os.IsNotExist(err) {
+		if errors.Is(err, fs.ErrNotExist) {
```
Comment (art2ip) [inline cmd/storageserv/config/config.go:168]: Missing ! operator in if stmt.

**Review by verapaypal — APPROVED** [review]
fix ! in review comments, otherwise look ok

**Review by verapaypal — DISMISSED** [review]
fix ! in review comments, otherwise look ok

File: cmd/storageserv/config/config.go (hunk 1 above)
Comment (NeetishPathak) [reply cmd/storageserv/config/config.go]: Add ! operator in if statement in latest push , rebased

File: cmd/storageserv/storage/db/dbcopy/main.go
[the remark this answers is not in the document]
Comment (NeetishPathak) [reply cmd/storageserv/storage/db/dbcopy/main.go:1]: Added ! now

File: cmd/storageserv/storage/db/dbcopy/main.go [hunk 2]
```diff
@@ -205,7 +207,7 @@ func main() {
 	newDir := fmt.Sprintf("%s/dbcopy_%s", filepath.Dir(dbPath), filepath.Base(dbPath))
 	if len(tgtPath) > 0 {
 		_, err := os.Stat(newDir)
-		if !os.IsNotExist(err) {
+		if errors.Is(err, fs.ErrNotExist) {
```
[the remark this answers is not in the document]
Comment (NeetishPathak) [reply cmd/storageserv/storage/db/dbcopy/main.go]: Updated 
https://github.com/paypal/junodb/pull/88/commits/33a67b51ef23028c61d7a9e6f855803781220172

File: cmd/storageserv/storage/db/dbcopy/config.go [hunk 3]
```diff
@@ -72,7 +74,7 @@ func LoadConfig(file string) string {
 	if !cfg.MicroShardsEnabled {
 		tagFile := fmt.Sprintf("%s/microshard_enabled.txt", dbPath)
 		_, err := os.Stat(tagFile)
-		if !os.IsNotExist(err) {
+		if errors.Is(err, fs.ErrNotExist) {
```
Comment (art2ip) [inline cmd/storageserv/storage/db/dbcopy/config.go:77]: missing ! in the if stmt.

Comment (NeetishPathak) [reply cmd/storageserv/storage/db/dbcopy/config.go]: Updated in https://github.com/paypal/junodb/pull/88/commits/945fb8bbd0d5c9a19a3cb2b6ee8e0d7aec0be7b4

**Review by yapingshi — COMMENTED** [review]
Looks good.

**Review by yapingshi — COMMENTED** [review]
Looks good.

**Review by yapingshi — APPROVED** [review]

**Review by jostanislas — APPROVED** [review]
LGTM

**Review by art2ip — APPROVED** [review]
Looks good.

**Pull request merged**